src/net_processing.cpp

```diff
@@ -425,6 +428,7 @@ class PeerManagerImpl final : public PeerManager
     ChainstateManager& m_chainman;
     CTxMemPool& m_mempool;
     TxRequestTracker m_txrequest GUARDED_BY(::cs_main);
+    TxReconciliationTracker m_reconciliation;
```

Comment:
in d53b4bc0360760737b250b81e82c4d8e6e6b1144:

Perhaps it would make more sense to have `PeerManagerImpl` keep a `std::unique_ptr<TxReconciliationTracker>` and *only* initialize the recon tracker if `-txrecon` is on?
And further, check whether `m_reconciliation` points to something to figure out whether we have txrecon enabled, instead of querying the ArgsManager every time.